src/javascript/app_2/Modules/Trading/Components/Elements/purchase-button.jsx

```diff
@@ -0,0 +1,59 @@
+import classNames        from 'classnames';
+import React             from 'react';
+import { localize }      from '_common/localize';
+import Button            from 'App/Components/Form/button.jsx';
+import Money             from 'App/Components/Elements/money.jsx';
+import { IconTradeType } from 'Assets/Trading/Types';
+
+const PurchaseButton = ({
+    currency,
+    info,
+    is_contract_mode,
+    is_disabled,
+    is_high_low,
+    is_loading,
+    onClickPurchase,
+    trade_types,
+    type,
+}) => (
+    <Button
+        is_disabled={ is_contract_mode || is_disabled }
+        id={`purchase_${type}`}
+        className={classNames(
+            'btn--primary',
+            'btn-purchase',
+            { 'animate': is_loading })}
+        has_effect
+        onClick={() => { onClickPurchase(info.id, info.stake, type); }}
+    >
+        <React.Fragment>
+            <div className='btn-purchase__content'>
+                <div className='btn-purchase__trade-type'>
+                    <div className='btn-purchase__icon'>
+                        <IconTradeType
+                            type={!is_disabled ? !is_high_low ? type.toLowerCase() : `${type.toLowerCase()}_barrier` : ''}
```

Comment:
please use `(` to highlight the related nested, browsers can get confused at times